### ISSUE TITLE
Command-line build failing due to missing targets in microsoft.sql-database-projects on Ubuntu

### ISSUE THREAD

**Issue by MHacker9404** [microsoft/azuredatastudio#14075]
- Azure Data Studio Version: 1.25.2

Steps to Reproduce:

1.Created a buildspec for AWS CodeBuild that installs ADS - screenshot attached 
![buildspec](https://user-images.githubusercontent.com/4184348/106049043-ac3ca480-60b3-11eb-83b3-c542667334e4.png)
2. When the build executes - the command-line build fails with the error message that the imported project "...microsoft.sql-database-projects-0.5.1/BuildDirectory/Microsoft.Data.Tools.Schema.SqlTasks.targets" was not found."
![error result](https://user-images.githubusercontent.com/4184348/106049168-d9895280-60b3-11eb-9142-81d000cd57fe.png)

**Comment by malcolm738** [comment]
I think is is the cross platform build issue I had
Using the reference - https://github.com/rr-wfm/MSBuild.Sdk.SqlProj
I updated to add Sdk ref to the .sqlproj file and "MSBuild.Sdk.SqlProj/1.10.0" works for me 
<Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" ToolsVersion="4.0" Sdk="MSBuild.Sdk.SqlProj/1.10.0">

Tests as build ADS build on windows
Also with dotnet build command in linux container
Base Image: mcr.microsoft.com/mssql/server base 
Dot Net: aspnetcore-runtime-5.0 dotnet-sdk-5.0

The update appears to fix the resolution issues with;
  <Import Condition="'$(NetCoreBuild)' == 'true'" Project="$(NETCoreTargetsPath)\Microsoft.Data.Tools.Schema.SqlTasks.targets"/>
  <Import Condition="'$(NetCoreBuild)' != 'true' AND '$(SQLDBExtensionsRefPath)' != ''" Project="$(SQLDBExtensionsRefPath)\Microsoft.Data.Tools.Schema.SqlTasks.targets" />
  <Import Condition="'$(NetCoreBuild)' != 'true' AND '$(SQLDBExtensionsRefPath)' == ''" Project="$(MSBuildExtensionsPath)\Microsoft\VisualStudio\v$(VisualStudioVersion)\SSDT\Microsoft.Data.Tools.Schema.SqlTasks.targets" />

**Comment by malcolm738** [comment]
Just double checked and with the following updates to the sqlproj file i can build from ADS on windows
and with dotnet build once code shipped into ubuntu 18.04 container

Added Sdk ref, TargetFramework and SqlServerVersion
Also updated the Import of .targets to check '$(NetCoreBuild)'
is 'true'
or is not 'true' and not '' (empty)

< !-- File updates applied based on https://github.com/rr-wfm/MSBuild.Sdk.SqlProj -->
< Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" ToolsVersion="4.0" Sdk="MSBuild.Sdk.SqlProj/1.11.0">
< !-- Start File updates, required by .net Core build -->
    < TargetFramework>netcoreapp2.1</TargetFramework>
    < SqlServerVersion>SqlAzure</SqlServerVersion>
< !-- End File updates, required by .net Core build -->
< !-- Start File updates, check '$(NetCoreBuild)' is not 'true' and not empty '', is just not 'true' linux missing reference error -->
  < Import Condition="'$(NetCoreBuild)' != 'true' AND '$(NetCoreBuild)' != '' AND '$(SQLDBExtensionsRefPath)' != ''" Project="$(SQLDBExtensionsRefPath)\Microsoft.Data.Tools.Schema.SqlTasks.targets"/>
  < Import Condition="'$(NetCoreBuild)' != 'true' AND '$(NetCoreBuild)' != '' AND '$(SQLDBExtensionsRefPath)' == ''" Project="$(MSBuildExtensionsPath)\Microsoft\VisualStudio\v$(VisualStudioVersion)\SSDT\Microsoft.Data.Tools.Schema.SqlTasks.targets"/>
< !-- End File updates, check '$(NetCoreBuild)' is not 'true' and not empty '', is just not 'true' linux missing reference error -->

**Comment by kisantia** [comment]
looks similar to the problem in #13409.

@MHacker9404 the targets file gets copied from SqlToolsService during the first build using ADS, so that needs to be done before command line build will work.

**Comment by malcolm738** [comment]
@MHacker9404 yes chicken and egg, the hack's I made to get working for me provide the farmer (build engine) clues on where the hen house (.target files) so they can collect eggs from their chickens
I believe the updates help the command line build for net core (5 in my case) to do the build without the need for Visual Studio, VSCode or ADS to be installed.
However the hack needs to be validated and have the .sqlproj file managed by the ADS extension to be aligned to the build engine functionality
Also probable needs to handle 'SqlServerVersion' as I can't see this in the default file so don't know the target SQL version

**Comment by malcolm738** [comment]
Update after testing and changes

**Use of the 'Sdk="MSBuild.Sdk.SqlProj/1.10.0"'**
This allows a bypass of the .target file and will by default build all '.sql' files in the projects subfolders into a DacPac
Limited support exists for referencing 'other' databases, this requires Nuget version so has issues with the system database
Where I clone and extend the database in my project (Unit Tests on top of base database) this requires a hack of the .sqlproj file
So using the Sdk option will work for single standalone databases in most cases with only the above hacks

**Going back to the default .sqlproj**
 I needed an couple of updates;
_**dotnet build {project path} /p:NetCoreBuild=true /p:NETCoreTargetsPath="{BuildDirectory}"**_
For {BuildDirectory} I cloned the folder "C:\Users\ {user} \ .azuredatastudio\extensions\microsoft.sql-database-projects-0.6.1\BuildDirectory" into my build machine

I also notice the project extension has an Add New or Delete/Exclude file option only, with some of my code, like referenced PostDeploy scripts
I needed
< None Include="Scripts\Post\Script1.sql" / >
Instead of 
< Build Include="Scripts\Post\Script1.sql" / >

I also had some issues while testing with the bin and obj folders being picked up
So for build folders and pre/post deployment scripts added;
    < Content Remove="bin\ * * \ * . *" / >
    < Content Remove="obj\ * * \ * . *" / >
    < Folder Include="Scripts" / >
    < PreDeploy Include="Scripts\PreDeployment.sql" / >
    < Content Remove="Scripts\Pre\ * . *" / >
    < PostDeploy Include="Scripts\PostDeployment.sql" / >
    < Content Remove="Scripts\Post\ * . *" / >
Then managed the child pre/post scrips via the workspace

So some basic hacks for default .sqlproj and updated reference on build machine are now working
Nice to have updates
Be able to extract the BuildDirectory from the extension or separate download for addition to Build Machine

Addition of default exclusion of files in system folders like the build folders and error handling to prevent explicit use of those folder

Addition on menu to amend Build Option, Build / None for the included scripts, or when excluding a file setting to None so it's still visible and maybe set text color to silver

**Comment by kisantia** [comment]
@malcolm738 can you open up separate issues for these enhancement requests for easier tracking? Thanks!

**Comment by malcolm738** [comment]
Hey @kisantia no problem, I already added a note to https://github.com/microsoft/azuredatastudio/issues/14383 for the files in  project to be excluded.
I will raise something new for each option so they are recorded
I understand that support for the 'MSBuild.Sdk.SqlProj' may be closed as an uncontrolled third party

**Comment by RoozbehBandpey** [comment]
Easiest way to build DacPac file on a linux agent is done via [MSBuild.Sdk.SqlProj](https://github.com/rr-wfm/MSBuild.Sdk.SqlProj)
Go to your database project directory in parallel to .sqlproj file create a directory like `DB.Build` under it create `DB.Build.csproj` copy.pase the content as below
```xml
<Project Sdk="MSBuild.Sdk.SqlProj/1.1.0"> <!-- This will pull in the required tools and dependencies to build a .dacpac with .NET Core -->
	<PropertyGroup>
		<TargetFramework>netstandard2.0</TargetFramework>
	</PropertyGroup>
	<ItemGroup>
		<Content Include="..\src\DB\masterdata\**\*.sql" /> <!-- link in the new .csproj to the .sql scripts in your existing database project -->
	</ItemGroup>
</Project>
```

After run you will see dacpac file appears under `DB.Build/bin/Release/netstandard2.0/DB.Build.dacpac`

Here's my build agent output (Ubuntu agent on Azure devops)
```bash
Starting: SQL DB build Release
==============================================================================
Task         : .NET Core
Description  : Build, test, package, or publish a dotnet application, or run a custom dotnet command
Version      : 2.187.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/build/dotnet-core-cli
==============================================================================
Info: .NET Core SDK/runtime 2.2 and 3.0 are now End of Life(EOL) and have been removed from all hosted agents. If you're using these SDK/runtimes on hosted agents, kindly upgrade to newer versions which are not EOL, or else use UseDotNet task to install the required version.
/opt/hostedtoolcache/dotnet/dotnet build /home/vsts/work/1/s/src/RecommenderAPI.DB/RecommenderAPI.DB/RecommenderAPI.DB.Build/RecommenderAPI.DB.Build.csproj -dl:CentralLogger,"/home/vsts/work/_tasks/DotNetCoreCLI_5541a522-603c-47ad-91fc-a4b1d163081b/2.187.0/dotnet-build-helpers/Microsoft.TeamFoundation.DistributedTask.MSBuild.Logger.dll"*ForwardingLogger,"/home/vsts/work/_tasks/DotNetCoreCLI_5541a522-603c-47ad-91fc-a4b1d163081b/2.187.0/dotnet-build-helpers/Microsoft.TeamFoundation.DistributedTask.MSBuild.Logger.dll" --configuration Release /p:DeployOnBuild=true /p:WebPublishMethod=Package /p:PackageAsSingleFile=true /p:SkipInvalidConfigurations=true /p:PackageLocation=/home/vsts/work/1/recommender-service-cicd/DacPac/
Microsoft (R) Build Engine version 16.5.0+d4cbfca49 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 51.72 ms for /home/vsts/work/1/s/src/RecommenderAPI.DB/RecommenderAPI.DB/RecommenderAPI.DB.Build/RecommenderAPI.DB.Build.csproj.
  Using package name RecommenderAPI.DB.Build and version 1.0.0
  Using SQL Server version Sql150
  Deleting existing file /home/vsts/work/1/s/src/RecommenderAPI.DB/RecommenderAPI.DB/RecommenderAPI.DB.Build/obj/Release/netstandard2.0/RecommenderAPI.DB.Build.dacpac
  Writing model to /home/vsts/work/1/s/src/RecommenderAPI.DB/RecommenderAPI.DB/RecommenderAPI.DB.Build/obj/Release/netstandard2.0/RecommenderAPI.DB.Build.dacpac
  RecommenderAPI.DB.Build -> /home/vsts/work/1/s/src/RecommenderAPI.DB/RecommenderAPI.DB/RecommenderAPI.DB.Build/bin/Release/netstandard2.0/RecommenderAPI.DB.Build.dacpac

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.71

Finishing: SQL DB build Release

```

>Note: Make sure to restore you NuGet packages in step prior to build

**Comment by kisantia** [comment]
An alpha package of Microsoft.Build.Sql has been released if you'd like to try it and give feedback: https://www.nuget.org/packages/Microsoft.Build.Sql/0.1.1-alpha

instructions can be found here: https://github.com/microsoft/DacFx/tree/main/src/Microsoft.Build.Sql

**Comment by kisantia** [comment]
Microsoft.Build.Sql is now public preview and SDK-style projects are supported in ADS! More info: https://techcommunity.microsoft.com/t5/azure-sql-blog/microsoft-build-sql-the-next-frontier-of-sql-projects/ba-p/3290628
